crates/bevy_scene/src/scene_spawner.rs

```diff
@@ -58,6 +63,28 @@ impl SceneSpawner {
         Ok(())
     }
 
+    pub fn unload_sync(
+        &mut self,
+        world: &mut World,
+        scene_handle: Handle<Scene>,
+    ) -> Result<(), SceneSpawnError> {
+        if let Some(instance_ids) = self.spawned_scenes.get(&scene_handle) {
+            for instance_id in instance_ids {
+                if let Some(instance) = self.spawned_instances.get(&instance_id) {
+                    for entity in instance.entity_map.values() {
+                        let _ = world.despawn(*entity); // Ignore the result, unload only cares if it exists.
+                    }
+                }
+            }
+
+            self.loaded_scenes
```

Comment:
lets replace these retain() calls with
```rust
self.loaded_scenes.remove(&scene_handle);
```

These are hashsets/hashmaps, which means we don't need to iterate over the whole collection to find a key.